diff -Naur which-2.21/which.info which-2.21.tpg/which.info
--- which-2.21/which.info	2015-03-20 16:30:22.000000000 +0000
+++ which-2.21.tpg/which.info	2015-03-21 10:07:38.595967016 +0000
@@ -5,6 +5,10 @@
 * Which: (which).                               Show full path of commands.
 END-INFO-DIR-ENTRY
 
+INFO-DIR-SECTION Utilities
+START-INFO-DIR-ENTRY
+* GNU/Which: (which).   The which command.
+END-INFO-DIR-ENTRY
 
 This file documents 'which' version 2.x.
 
