diff -up which-2.23/tilde.c.1~ which-2.23/tilde.c
--- which-2.23/tilde.c.1~	2025-02-06 23:50:42.461701496 +0100
+++ which-2.23/tilde.c	2025-02-06 23:51:22.350420962 +0100
@@ -189,7 +189,8 @@ char *tilde_expand(const char *string)
   int result_size, result_index;
 
   result_index = result_size = 0;
-  if (result = strchr(string, '~'))
+  result = strchr(string, '~');
+  if (result)
     result = (char *) xmalloc(result_size = (strlen(string) + 16));
   else
     result = (char *) xmalloc(result_size = (strlen(string) + 1));
diff -up which-2.23/which.c.1~ which-2.23/which.c
--- which-2.23/which.c.1~	2025-02-06 23:50:42.462094637 +0100
+++ which-2.23/which.c	2025-02-06 23:52:04.813018438 +0100
@@ -734,6 +734,8 @@ int main(int argc, char *argv[])
   if (path_list)
     free(path_list);
 #endif
+  if (abs_path)
+    free(abs_path);
   return fail_count;
 }
 
